src/OpenLoco/StationManager.cpp

```diff
@@ -124,22 +128,210 @@ namespace OpenLoco::StationManager
+                if (surface == nullptr)
+                    continue;
+
+                auto townHeightDiff = (position.z / 4) - surface->baseZ();
+                if (townHeightDiff > 20)
+                {
+                    // Heights
+                    if ((realNamesInUse & (1 << 9)) == 0)
+                        return StringManager::toTownName(StringIds::station_town_heights);
+                }
+                else if (townHeightDiff < -20)
+                {
+                    // Valley
+                    if ((realNamesInUse & (1 << 8)) == 0)
+                        return StringManager::toTownName(StringIds::station_town_valley);
+                }
+                break;
+            }
+        }
+
+        // 0x0048FB8B
+        if ((realNamesInUse & (1 << 0)) == 0)
+            return StringManager::toTownName(StringIds::station_town);
+
+        auto town = TownManager::get(townId);
+        {
+            auto xDiff = std::abs(position.x - town->x);
+            auto yDiff = std::abs(position.y - town->y);
+            if (xDiff + yDiff <= 120)
+            {
+                // Central
+                if ((realNamesInUse & (1 << 5)) == 0)
+                    return StringManager::toTownName(StringIds::station_town_central);
+            }
+        }
+
+        if (position.x <= town->x && position.y <= town->y)
+        {
+            // North
+            if ((realNamesInUse & (1 << 1)) == 0)
+                return StringManager::toTownName(StringIds::station_town_north);
+        }
+
+        if (position.x >= town->x && position.y >= town->y)
+        {
+            // South
+            if ((realNamesInUse & (1 << 2)) == 0)
+                return StringManager::toTownName(StringIds::station_town_south);
+        }
+
+        if (position.x <= town->x && position.y >= town->y)
+        {
+            // East
+            if ((realNamesInUse & (1 << 3)) == 0)
+                return StringManager::toTownName(StringIds::station_town_east);
+        }
+
+        if (position.x >= town->x && position.y <= town->y)
+        {
+            // West
+            if ((realNamesInUse & (1 << 4)) == 0)
+                return StringManager::toTownName(StringIds::station_town_west);
+        }
+
+        // 0x0048FC5C
+        string_id foundName = StringIds::empty;
+        for (auto i = 5; i < 47; i++)
+        {
+            // At several positions, skip the entries we've already tested for earlier.
+            if (i == 8)
+                i = 12;
+            else if (i == 13)
+                i = 17;
+            else if (i == 22)
+                i = 24;
+
+            bool realNameIsInUse = (i < 27 && (realNamesInUse & (1 << i)) != 0);
+            bool ordinalNameIsInUse = (i >= 27 && (ordinalNamesInUse & (1 << (i - 27))) != 0);
+            if (!(realNameIsInUse || ordinalNameIsInUse))
+            {
+                foundName = StringManager::toTownName(StringIds::station_town + i);
+                break;
```

Comment:
Why even break here. Just return the value then you dont need to have the foundName variable. All a bit moot though if you change the loop as per previous comment.